src/tools/web/client/src/components/TreeView.jsx

```diff
@@ -0,0 +1,72 @@
+// adapted from https://github.com/chenglou/react-treeview
```

Comment:
No other comments in the file. What did you change? Why not use the treeview? Is yours different enough to claim authorship? (Otherwise we have license troubles)